### PR TITLE
fix: redeploy all modules on schema changes

### PR DESCRIPTION
**Pull request by wesbillman**
When the `ftl dev` command detects a schema change in any module, it will trigger a rebuild of all modules, including the one that caused the schema change.

This isn't ideal because we might end up rebuilding a single module twice (if it caused a schema change), but the logic is simple and rebuilds are fast.

### CODE REVIEW

File: cmd/ftl/cmd_dev.go [hunk 1]
```diff
@@ -38,6 +41,24 @@ func (d *devCmd) Run(ctx context.Context, client ftlv1connect.ControllerServiceC
 	d.modules = make(map[string]moduleFolderInfo)
 	d.client = client
 
+	stream, err := client.PullSchema(ctx, connect.NewRequest(&ftlv1.PullSchemaRequest{}))
```
Comment (alecthomas) [inline cmd/ftl/cmd_dev.go:44]: This should be in the goroutine itself, and should handle disconnects of the stream. ie. there should be _two_ loops, an outer one initiating the stream and an inner one pulling from the stream. Don't forget to close the stream if it errors.

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
Rebuilds are fast with Go, but not so much with Kotlin, so we should be smarter about this in future. Can you create a ticket to remind us to fix this?

**Comment by wesbillman** [comment]
> Rebuilds are fast with Go, but not so much with Kotlin, so we should be smarter about this in future. Can you create a ticket to remind us to fix this?

https://github.com/TBD54566975/ftl/issues/792

File: cmd/ftl/cmd_dev.go [hunk 2]
```diff
@@ -74,13 +108,44 @@ func (d *devCmd) Run(ctx context.Context, client ftlv1connect.ControllerServiceC
...
+	logger := log.FromContext(ctx)
+	for {
+		stream, err := client.PullSchema(ctx, connect.NewRequest(&ftlv1.PullSchemaRequest{}))
+		if err != nil {
+			logger.Errorf(err, "Error connecting to FTL. Will retry...")
+			time.Sleep(d.ReconnectDelay)
+			continue
+		}
+
+		for {
```
Comment (alecthomas) [inline cmd/ftl/cmd_dev.go:131]: A better pattern is:

```
for stream.Receive() {
  msg := stream.Msg()
 // ..
}
stream.Close()
// ...
```

**Review by alecthomas — APPROVED** [review]

**Pull request merged**